packages/datatrak-web/src/views/AccountSettingsPage/PersonalDetailsSection.tsx

```diff
@@ -4,10 +4,168 @@
  */
 
 import React from 'react';
+import styled from 'styled-components';
+import { useForm, Controller, SubmitHandler } from 'react-hook-form';
 import { AccountSettingsSection } from './AccountSettingsSection';
+import { Button } from '../../components';
+import { successToast } from '../../utils';
+import { TextField } from '@tupaia/ui-components';
+import { useCurrentUser } from '../../api';
+import { useEditUser } from '../../api/mutations';
+import { UserAccountDetails } from '../../types';
+
+type PersonalDetailsFormFields = Pick<
+  UserAccountDetails,
+  'firstName' | 'lastName' | 'employer' | 'position' | 'mobileNumber'
+>;
+
+const ButtonWrapper = styled.div`
+  grid-column: -2;
+`;
+
+const StyledTextField = styled(TextField)`
+  margin: 0; // Use gap on parent to control spacing
+`;
+
+const PersonalDetailsForm = styled.form`
+  display: grid;
+  gap: 1.56rem 1.25rem;
+  max-width: 44.25rem;
+  width: 100%;
+  ${({ theme }) => theme.breakpoints.up('sm')} {
+    grid-template-columns: repeat(2, 1fr);
+  }
+
+  .MuiFormLabel-root {
+    color: ${props => props.theme.palette.text.primary};
+    font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
+  }
+`;
 
 export const PersonalDetailsSection = () => {
+  const { mutate: updateUser } = useEditUser(() => successToast('Personal details updated'));
+  const user = useCurrentUser();
+
+  const {
+    control,
+    formState: { isDirty, dirtyFields },
+    handleSubmit,
+    register,
+  } = useForm<PersonalDetailsFormFields>({
+    defaultValues: {
+      firstName: user.firstName ?? '',
+      lastName: user.lastName ?? '',
+      mobileNumber: user.mobileNumber ?? '',
+      employer: user.employer ?? '',
+      position: user.position ?? '',
+    },
+  });
+
+  function onSubmit(userDetails): SubmitHandler<PersonalDetailsFormFields> {
```

Comment:
I’d prefer to restrict the type of `details` to `UserAccountDetails` here (and on line 77), but for some bizarre reason if I do either, then the form submission simply does not work. Baffling because when I let TypeScript infer the type, it gives me `UserAccountDetails`

If you have some idea of why this is, I’d love to know!

Reply:
This will be resolved in an upcoming PR